app/models/exchange_rate_currency_rate.rb

```diff
@@ -20,6 +22,12 @@ class ExchangeRateCurrencyRate < Sequel::Model
 
   content_addressable_fields :currency_code, :validity_start_date, :validity_end_date, :country, :country_code
 
+  def validate
+    super
+
+    validates_unique(%i[currency_code validity_start_date validity_end_date rate_type])
```

Comment:
This validation is to avoid adding the same rate for the same period, dates and types more than once.

Reply:
Period wise do we need to include both start and end? considering so many are nil?

Reply:
Don't we already handle this at the database level?